modules/icgc-argo-workflows/prep/metrics/resources/usr/bin/dnaaln.py

```diff
@@ -0,0 +1,167 @@
+      'sample_id': sampleId,
+      'metrics': {}
+    }
+    for f in sorted(glob(multiqc+'/*.txt')):
+      for tool_metrics in tool_fieldmap.keys():
+        if f.endswith(tool_metrics+'.txt'):
+          with open(f, 'r') as fn: 
+            mqc_stats[tool_metrics] = []
+            reader = csv.DictReader(fn, delimiter="\t")
+            for row in reader:
+              if not sampleId in row.get('Sample'): continue
+              mqc_stats[tool_metrics].append(row)
+              for ftype in tool_fieldmap.keys():
+                if not ftype == tool_metrics: continue
+                for f1,f2 in tool_fieldmap[ftype].items():
+                  mqc_stats['metrics'][f1] = round(float(row.get(f2)), 4)
+
+    # convert the fraction to percentage for given fields
+    for fn in fra2pct_fields:
+      if fn not in mqc_stats['metrics']: continue # <if not mqc_stats['metrics'].get(fn): continue> is wrong as it is possible mqc_stats['metrics'][fn] is 0.0 (it will return false by mistake)
+      new_value = round(float(mqc_stats['metrics'][fn]) * 100, 2)
+      mqc_stats['metrics'].update({
+        fn: new_value
+      })
+
+    # change type to integer for given fields
+    for fn in integer_fields:
+        if fn not in mqc_stats['metrics']: continue
+        new_value = int(mqc_stats['metrics'][fn])
+        mqc_stats['metrics'].update({
+          fn: new_value
+        })
+    
+    return mqc_stats
+
+def main():
+    """
+    Python implementation of tool: prep_metrics.py
+    """
+
+    parser = argparse.ArgumentParser(description='Tool: prep_metrics')
+    parser.add_argument("-s", "--sampleId", dest="sampleId", required=True, help="Input sampleId", type=str)
+    parser.add_argument("-m", "--multiqc", dest="multiqc", required=True, help="multiqc files folder")
+    parser.add_argument("-q", "--qc_files", dest="qc_files", required=True, type=str, nargs="+", help="qc files")
+
+    args = parser.parse_args()
+    
+    # get tool_specific & aggregated metrics from multiqc
+    mqc_stats = {}
+    if args.multiqc:
+      mqc_stats = get_mqc_stats(args.multiqc, args.sampleId)
+
+    # get tool_specific & aggregated metrics from qc_files when they're not retrieved by multiqc
+    for fn in sorted(args.qc_files):
+      # GATK4 calculateContamination
+      if fn.endswith('contamination.table'):
+        file_type = 'gatk_contamination'
+        if file_type in mqc_stats: continue
+        mqc_stats[file_type] = []
+        with open(fn, 'r') as f:     
+          reader = csv.DictReader(f, delimiter="\t")
+          for row in reader:
+            mqc_stats[file_type].append(row)
+            mqc_stats['metrics'].update({
+              'cross_contamination_rate': float(row.get('contamination')),
+              'cross_contamination_error': float(row.get('error'))
+            })
+
+    mqc_stats_updated = {k: v for k, v in mqc_stats.items() if v}
+
+    with open("%s.dnaaln_argo_metrics.json" % (args.sampleId), 'w') as f:
```

Comment:
You should use `%s.dnaaln.argo_metrics.json`, change the `_` to `.`
Otherwise cause the error in tests:
```
Missing output file(s) `*.argo_metrics.json` expected by process
```